tests/integration/test_zos_files.py

```diff
@@ -29,9 +29,32 @@ def setUp(self):
 
     def test_list_dsn_should_return_a_list_of_datasets(self):
         """Executing list_dsn method should return a list of found datasets."""
-        command_output = self.files.list_dsn(self.files_fixtures["TEST_HLQ"])
-        self.assertIsInstance(command_output['items'], list)
-
+        
+        scenarios = [
+            {"attributes": False, "expected_attributes": ["dsname"]},
+            {"attributes": True, "expected_attributes": ["dsname", "dsorg"]}
```

Comment:
Sorry for being picky, but could we use something that will always be there, no matter the state of the dataset
```jsonc
// Example of archived/migrated datasets
{
  "items": [
    {
      "dsname": "USERID.CLASS.C",
      "migr": "YES",
      "vol": "MIGRAT"
    }
  ],
  "returnedRows": 1,
  "JSONversion": 1
}
// Example of VSAM datasets 
{
  "items": [
    {
      "dsname": "USERID.PUBLIC.TEST.ZFS",
      "catnm": "ICF.LPAR001",
      "dsorg": "VS",
      "migr": "NO",
      "mvol": "N",
      "vol": "*VSAM*",
      "vols": "*VSAM*"
    },
    {
      "dsname": "USEIRD.PUBLIC.TEST.ZFS.DATA",
      "blksz": "?",
      "catnm": "ICF.LPAR001",
      "cdate": "2020/08/12",
      "dev": "3390",
      "dsorg": "VS",
      "edate": "***None***",
      "extx": "1",
      "lrecl": "?",
      "migr": "NO",
      "mvol": "N",
      "ovf": "NO",
      "rdate": "2020/08/12",
      "recfm": "?",
      "sizex": "1410",
      "spacu": "CYLINDERS",
      "used": "?",
      "vol": "TSOA20",
      "vols": "TSOA20"
    }
  ],
  "returnedRows": 2,
  "JSONversion": 1
}

Reply:
It seems that both "vol" and "migr" attributes are present in both cases. Therefore, I will need to modify the current implementation and possibly remove the assertion of checking "dorg" since it does not appear in both cases.